src/main/java/com/codegym/jrugotom5/service/AdvertService.java

```diff
@@ -41,7 +50,22 @@ public List<AdvertFullInfoDTO> getAdvertsByDateRange(LocalDate from, LocalDate t
 
     public List<AdvertBasicInfoDTO> getAllAdverts() {
         return Streamable.of(advertRepository.findAll())
-                .map(advert->modelMapper.map(advert, AdvertBasicInfoDTO.class))
+                .map(advert -> modelMapper.map(advert, AdvertBasicInfoDTO.class))
                 .toList();
     }
+
+    @Transactional
+    public AdvertFullInfoDTO createAdvert(AdvertCreateDTO advertCreateDTO) {
+        Long id = advertCreateDTO.getUserCreatorId();
+        if (!userService.userExistsById(id)) {
+            log.error("Could not find User with id {}", id);
+            throw new UserNotFoundException("User with id " + id + " does not exist.");
+        }
+        Advert advertEntity = modelMapper.map(advertCreateDTO, Advert.class);
```

Comment:
Your mapper conversion doesn't work! Try to debug and check "ID" value of advertEntity.
On this step it must have 'null' value.
It seems `modelMapper` converts `userCreatorId` to `id` of advert entity.

Also i suppose you should set userCreatorId for your resulting DTO AdvertFullInfoDTO manually after mapping.